backend/src/exceptions.py

```diff
@@ -0,0 +1,3 @@
+class AuthenticationError(Exception):
```

Comment:
I like this but I think we should change the name to include "RefreshToken" somewhere
Since we only want to use it when the refresh token is invalid the current name is a bit misleading, its not a general authentication issue